lib.rs

```diff
@@ -93,3 +93,846 @@ pub mod src {
+        }
+
+        pub fn set_decode_frame_type(&mut self, decode_frame_type: DecodeFrameType) {
+            self.dav1d_settings.decode_frame_type = decode_frame_type.into();
+        }
+
+        pub fn get_decode_frame_type(&self) -> DecodeFrameType {
+            DecodeFrameType::try_from(self.dav1d_settings.decode_frame_type)
+                .expect("Invalid Dav1dDecodeFrameType")
+        }
+    }
+
+    bitflags::bitflags! {
+        #[derive(Copy, Clone, Debug, PartialEq, Eq, Default)]
+        pub struct InloopFilterType: u32 {
+            const DEBLOCK = DAV1D_INLOOPFILTER_DEBLOCK;
+            const CDEF = DAV1D_INLOOPFILTER_CDEF;
+            const RESTORATION = DAV1D_INLOOPFILTER_RESTORATION;
+        }
+    }
+
+    #[derive(Default, Debug, Copy, Clone, PartialEq, Eq)]
+    pub enum DecodeFrameType {
+        #[default]
+        All,
+        Reference,
+        Intra,
+        Key,
+    }
+
+    impl TryFrom<u32> for DecodeFrameType {
+        type Error = TryFromEnumError;
+
+        fn try_from(value: u32) -> Result<Self, Self::Error> {
+            match value {
+                DAV1D_DECODEFRAMETYPE_ALL => Ok(DecodeFrameType::All),
+                DAV1D_DECODEFRAMETYPE_REFERENCE => Ok(DecodeFrameType::Reference),
+                DAV1D_DECODEFRAMETYPE_INTRA => Ok(DecodeFrameType::Intra),
+                DAV1D_DECODEFRAMETYPE_KEY => Ok(DecodeFrameType::Key),
+                _ => Err(TryFromEnumError(())),
+            }
+        }
+    }
+
+    impl From<DecodeFrameType> for u32 {
+        fn from(v: DecodeFrameType) -> u32 {
+            match v {
+                DecodeFrameType::All => DAV1D_DECODEFRAMETYPE_ALL,
+                DecodeFrameType::Reference => DAV1D_DECODEFRAMETYPE_REFERENCE,
+                DecodeFrameType::Intra => DAV1D_DECODEFRAMETYPE_INTRA,
+                DecodeFrameType::Key => DAV1D_DECODEFRAMETYPE_KEY,
+            }
+        }
+    }
+
+    /// The error type returned when a conversion from a C enum fails.
+    #[derive(Debug, Copy, Clone, PartialEq, Eq)]
+    pub struct TryFromEnumError(());
+
+    impl std::fmt::Display for TryFromEnumError {
+        fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+            fmt.write_str("Invalid enum value")
+        }
+    }
+
+    impl From<std::convert::Infallible> for TryFromEnumError {
+        fn from(x: std::convert::Infallible) -> TryFromEnumError {
+            match x {}
+        }
+    }
+
+    impl std::error::Error for TryFromEnumError {}
+
+    /// A `dav1d` decoder instance.
+    pub struct Decoder {
+        dec: Dav1dContext,
+        pending_data: Option<Dav1dData>,
+    }
+
+    unsafe extern "C" fn release_wrapped_data<T: AsRef<[u8]>>(
+        _data: *const u8,
+        cookie: Option<SendSyncNonNull<std::ffi::c_void>>,
+    ) {
+        let cookie = cookie.unwrap().as_ptr().as_ptr();
+        let buf = unsafe { Box::from_raw(cookie as *mut T) };
```

Comment:
We haven't enabled it in CI yet, but we have `#![deny(clippy::undocumented_unsafe_blocks)]` on.  All `unsafe` blocks should be documented with `// SAFETY` comments.